[Addon Dev] allow all of rollup's OutputOptions to be passed to addon.output()

NullVoxPopuli opened the pull request:
Update!

This PR adds pass-through support for all of rollup's output options:

```js
output: addon.output({ sourcemap: true });
```

Intellisense confirmed working, too:
![image](https://user-images.githubusercontent.com/199018/152454664-1e0f8fec-0c6a-4b90-a598-a321f97267a3.png)


--------------------
(old PR description)


I found myself wanting to provide sourcemaps for libraries.
`addon.output()` does not allow for that, but you could do:
```js
output: {
  ...addon.output(),
  sourcemap: true,
}
```

but, that's extra code on the consuming side, when it could just be 1 extra line of config in the `new Addon` part of the rollup config file.

There are _a lot_ of output options tho: https://rollupjs.org/guide/en/#outputoptions-object.
:shrug: 

Maybe instead, we can take any object as an argument to `.output()` and merge that in the rollup config?

rwjblue commented on packages/addon-dev/src/rollup.ts:
```diff
@@ -62,7 +90,12 @@ export class Addon {
   // their bundle names (the bundle names get generated by `publicEntrypoints`
   // above).
   output() {
```
Why not just take options here?

NullVoxPopuli replied:
I _just_ edited the PR to ask that same question. Happy to change if that's preferred.

To answer "why not":  I didn't think of it initially :upside_down_face:

NullVoxPopuli replied:
> I just edited the PR to ask that same question

I'm really upset at github for continuing to revert my PR descriptions :cry: 
![image](https://user-images.githubusercontent.com/199018/152428764-fbbe4790-9d87-40cc-83cb-97811a98bb6d.png)

NullVoxPopuli replied:
oh it's back now. idk what's going on with github :(

NullVoxPopuli replied:
advantage of passing options to output function was too high! (changed to that approach)

NullVoxPopuli commented:
I removed `sourcemap` from the constructor, kept the docs tho, and added _all_ output options to the output function. Looks kinda fun (intellisense test):

![image](https://user-images.githubusercontent.com/199018/152454399-1ebddb5b-2075-46f4-8577-abfe1aa23329.png)

kinda weird they don't have any description or anything on these, but that's work we PR to rollup to help everyone out

ef4 commented:
This PR comes down the different between:

```js
{
  // current
  output: { ...addon.output(),  sourceMap: true }
}
```

and

```js
{
  // proposed
  output: addon.output({ sourceMap: true })
}
```

They're extremely close, but the proposed option forces you to read the documentation for `addon.output()` to understand what's going to happen, whereas the current option is Just Javascript™️ and it's transparent what's happening. So I'm not a fan.

NullVoxPopuli commented:
aight, sounds good! 🥳
